pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java

```diff
@@ -101,10 +112,33 @@ protected InstanceResponseBlock getNextBlock() {
       responseMetaData
           .put(MetadataKey.SYSTEM_ACTIVITIES_CPU_TIME_NS.getName(), String.valueOf(systemActivitiesCpuTimeNs));
 
+      // TODO: Remove this once the SelectionOrderByOperator is modified to throw an exception to catch cases where
+      //       an MV column (identifier or via transform) is present on the order-by list
+      logAndEmitMetricForQueryHasMVSelectionOrderBy(intermediateResultsBlock);
+
       return instanceResponseBlock;
     } else {
-      return new InstanceResponseBlock(getCombinedResults());
+      // TODO: Remove this once the SelectionOrderByOperator is modified to throw an exception to catch cases where
+      //       an MV column (identifier or via transform) is present on the order-by list
+      IntermediateResultsBlock intermediateResultsBlock = getCombinedResults();
+      logAndEmitMetricForQueryHasMVSelectionOrderBy(intermediateResultsBlock);
+
+      return new InstanceResponseBlock(intermediateResultsBlock);
+    }
+  }
+
+  private void logAndEmitMetricForQueryHasMVSelectionOrderBy(IntermediateResultsBlock intermediateResultsBlock) {
+    if (!intermediateResultsBlock.isQueryHasMVSelectionOrderBy()) {
+      return;
+    }
+
+    String tableName = _queryContext.getTableName();
+    if (_serverMetrics != null) {
+      _serverMetrics.addMeteredTableValue(tableName, ServerMeter.QUERY_HAS_MV_SELECTION_ORDER_BY, 1);
     }
+    List<OrderByExpressionContext> orderByExpressions = _queryContext.getOrderByExpressions();
+    LOGGER.warn("Table {} has MV column in ORDER BY. Expressions: {}", tableName,
+        orderByExpressions == null ? "" : orderByExpressions);
```

Comment:
`orderByExpressions == null `check can be avoided because if that were the case, we should not have gone past line 132

Reply:
okay i'll remove that, thanks!